### ISSUE TITLE
Need to handle null global variables

### ISSUE THREAD

**Issue by sw-tt-jaydeepcharadva** [ExpressionEngine/ExpressionEngine#4113]
Hello,

Recently I have upgrade Expression Engine 7.2.14 to 7.4.2 with latest php version 8.2. 
I am facing the issue with standard global variables after upgrade Expression Engine.

Error:
**Deprecated
str_replace(): Passing null to parameter #2 ($replace) of type array|string is deprecated
ee/legacy/libraries/Template.php, line 3178
Severity: E_DEPRECATED**

It's occured due to null is not handled for global variable. I have fixed it by adding if condition in \system\ee\legacy\libraries\Template.php file. I am attaching the screenshot for your reference.

![image_480](https://github.com/ExpressionEngine/ExpressionEngine/assets/68269830/da4fefb4-b0ab-410b-b78a-acd78e346f8c)

Above Template.php is a legacy library file. So, I can not directely change to it. Can you please, provide the suggested solution in the next version of Expression Engine?

**Comment by intoeetive** [comment]
@sw-tt-jaydeepcharadva the suggested code change cannot be made, because it will break other things in the system.

Are you seeing the deprecation notice just on the CP homepage? If so, it is likely caused by one of the widgets. Can you try disabling those one by one and find out which is the one that's causing the issue? I would assume it's a custom widget using EE template from `pro-dashboard-widgets` group, and I'd like to see it's code.

**Comment by sw-tt-jaydeepcharadva** [comment]
Sure let me check and update you

**Comment by sw-tt-jaydeepcharadva** [comment]
@intoeetive I have dubug the code and found that the pro variable having null entries. Due to that it's throwing above mentioned error. Can you please give me a suggestion? or add the solution, which I provide you.

**Comment by intoeetive** [comment]
@sw-tt-jaydeepcharadva that could be something we fixed in 7.4.4 : https://github.com/ExpressionEngine/ExpressionEngine/pull/4082 - would you mind checking?

**Comment by sw-tt-jaydeepcharadva** [comment]
I upgrade it to 7.4.4 but still it's throwing same message

**Comment by intoeetive** [comment]
Can you lay out the steps needed to replicate the error? Or perhaps you have specific error message or trace that shows the source of the problem

**Comment by sw-tt-jaydeepcharadva** [comment]
I have upgrade Expression engine from 7.2.17 to 7.4.4 with latest php version 8.2. It throws error as mentioned in below screenshot after upgrade.

![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/68269830/0ba686da-bca5-4d0a-bbc7-5328bb3ccd68)

I have fixed it by adding if condition in \system\ee\legacy\libraries\Template.php file. I am attaching the screenshot for your reference.

![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/68269830/ccbb3aeb-5181-44aa-bada-bee9670a4c0c)

Let me know if you need more information about it from my end.

**Comment by robinsowell** [comment]
Hrm- it feels like the problem is the variables not being set, rather than the handling in templates.  I'm wondering how to replicate that.  And it may be I can't with a new install and it's due to some legacy where it saved null and shouldn't.

Can you tell where it's coming from in pro?

**Comment by sw-tt-jaydeepcharadva** [comment]
The pro variable who having null values at that place it throwing this error.

**Comment by robinsowell** [comment]
Right- what's the variable- what type of variable(s) are throwing null- maybe a screen cap?  I want to replicate it, see if I get the error, and if not, compare the database values.

**Comment by sw-tt-jaydeepcharadva** [comment]
A Pro variable is a custom variable and there are many pro variables in my project which has no value. In which it's throwing error.

**Comment by robinsowell** [comment]
They may have an empty string, which would be fine.  We want to know what's actually null.  You could likely see by var_dumping the variables that are empty where you had the conditional check for the replace?  We see what the variable is- find it in pro- I make the same type of variable and leave it without adding a value.  Then we see if mine also breaks.  If not, we can compare the field in the database, see if there's a difference there.  Maybe old ones went in as null and new ones don't- something like that.

**Comment by sw-tt-jaydeepcharadva** [comment]
Hi @robinsowell,

I have review the data with var_dump and I have found that many pro variables having value as **NULL** entries.

**Comment by sw-tt-jaydeepcharadva** [comment]
@robinsowell you there?

**Comment by sw-tt-jaydeepcharadva** [comment]
@intoeetive @robinsowell Can you please provide any input on above point I need to upgrade Expression engine on priority basis.

**Comment by sw-tt-jaydeepcharadva** [comment]
@intoeetive @robinsowell So, What I have observe that the database table name ```exp_global_variables``` have total 42 entries which has default value null. If I change it to any other value than like ```xyz``` or ```abc``` then it is working fine. So, I request you to provide the support for handling null entries for global variables.
